src/core/cache_utils.jl

```diff
@@ -0,0 +1,55 @@
+const ParamCacheKey = NamedTuple{(:stage, :type, :name), NTuple{3, Symbol}}
+
+make_cache_key(stage, type, name) = (stage = stage, type = type, name = name)
+
+# Priority for keeping data in cache to serve reads. Currently unused.
+IS.@scoped_enum CachePriority begin
+    LOW
+    MEDIUM
+    HIGH
+end
+
+# TODO: we need to make scoped_enum handle words ending in 'y'
```

Comment:
I've opened an issue to track this: https://github.com/NREL-SIIP/InfrastructureSystems.jl/issues/186